src/main/kotlin/com/coder/gateway/CoderRemoteConnectionHandle.kt

```diff
@@ -523,5 +522,25 @@ class CoderRemoteConnectionHandle {
 
     companion object {
         val logger = Logger.getInstance(CoderRemoteConnectionHandle::class.java.simpleName)
+        fun processSetupCommand(
+            output: () -> String,
+            ignoreSetupFailure: Boolean
+        ) {
+            try {
+                val errorText = output
+                    .invoke()
+                    .lines()
+                    .firstOrNull { it.contains(GATEWAY_SETUP_COMMAND_ERROR) }
+                    ?.let { it.substring(it.indexOf(GATEWAY_SETUP_COMMAND_ERROR) + GATEWAY_SETUP_COMMAND_ERROR.length).trim() }
+
+                if (!errorText.isNullOrBlank()) {
+                    throw Exception(errorText)
```

Comment:
in Java&Kotlin it is best to avoid throwing the generic exception, for many reasons - but some of them are that it's too generic, and probably the most important one: upstream code can be forced to catch all checked exceptions 

In other words we should be more specific and throw one of java/kotlin checked exceptions or we can create our own custom exception. IOException is probably good enough, or maybe something custom like CmdExecutionException.

Reply:
So I called it `CoderSetupCommandException` and provided handling when we show the dialog so it's clearly says what it's a setup command problem, not just an abstract exception

Reply:
<img width="788" alt="image" src="https://github.com/user-attachments/assets/f3a03325-ec25-459a-8b00-8cc21c34a95e" />